src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookEndpoint.java

```diff
@@ -44,7 +44,11 @@ public HttpResponse doTrigger(StaplerRequest request, StaplerResponse response)
             case MIRROR_SYNCHRONIZED_EVENT:
                 return processMirrorSynchronizedEvent(request);
             case PULL_REQUEST_OPENED_EVENT:
+            case PULL_REQUEST_MERGED_EVENT:
```

Comment:
There are a lot of places where you check each of the three events- MERGED, DECLINED and DELETED, despite the fact that I think Jenkins treats them all equivalently. Perhaps when the event is first marshalled, you could convert these to a generic `PULL_REQUEST_CLOSED`, so we don't need so many extra comparisons